### DISCUSSION TITLE
The "Create a new document" problem

### DISCUSSION THREAD

**Discussion by logan-anderson** [tinacms/tinacms#2686] [General]
### Video explanation

Pt1: https://www.loom.com/share/9dbb5d84df76425eb5c9bb0534230de4
Pt2: https://www.loom.com/share/94d9f1521002406b816cf58832100540

### Text explanation

Currently in our starters and documentation we promote the following path for newly created documents.

- use `fallback: 'blocking'`
- always return a 200 status code (even if the document does not exist) and conditionally load editable content.

The main issue here is that when a user goes to a page that does not exist they will get a 200 status code, this is might be fine for small companies but is not a good idea for SEO reasons as URL will never become stale. 

Ideally, we should support sites that want to return 400 status codes for invalid documents 

#### How do we support this?

- If they are only using one branch for editing content one way around this is to wait until the new site builds on that branch with that new path available (not ideal)
- If they are only using one branch for editing we could solve this issue (using read only tokens in getStaticProps)

If branching is being used this causes even more issues because there is no way of knowing on the backend what branch to use (This is an entirely frontend concern right now). The only way to solve this is to use Next.js preview mode and read only tokens. While this is an ideal solution is will likely be a bit of work and require code changes in tinacms. This will be a decent amount of work (starting with a research spike) and cause changes in documentation and code changes.

**Comment by llibdude** [comment]
You hit it in the last paragraph of the ticket but I am using read only tokens and `return { notFound: true }`.  I very much like this solution but I understand it would have a high impact on code/docs.  

Here is my code snippet:
```javascript
export const getStaticPaths: GetStaticPaths = async () => {
  const client = ExperimentalGetTinaClient()
  const campaignsListData = await client.getCampaignsList()

  return {
    paths: campaignsListData.data.getCampaignsList.edges.map((page) => ({
      params: { campaign: [page.node.sys.filename] }
    })),
    fallback: 'blocking'
  }
}

export const getStaticProps: GetStaticProps = async ({ params }) => {
  const relativePath = String(`${(params.campaign as string[]).join('/')}.mdx`)
  const client = ExperimentalGetTinaClient()
  let tinaProps = await client.getCampaignsDocument({
    relativePath
  })

  if (!Object.keys(tinaProps.data).length) {
    // use read only tokens to get live data
    // @ts-ignore
    const response = await fetch(
      `https://content.tinajs.io/content/${process.env.NEXT_PUBLIC_TINA_CLIENT_ID}/github/${process.env.VERCEL_GIT_COMMIT_REF}`,
      {
        method: 'POST',
        body: JSON.stringify({ query: tinaProps.query, variables: { relativePath } }),
        headers: {
          'X-API-KEY': process.env.NEXT_PUBLIC_TINA_READONLY_TOKEN,
          'Content-Type': 'application/json'
        }
      }
    )
    tinaProps = await response.json()
    // This page does not exist so 404
    if (!tinaProps.data) {
      return {
        notFound: true
      }
    }
  }

  return {
    props: {
      ...tinaProps,
      revalidate: 60
    }
  }
}
```
